src/nvim/os/env.c

```diff
@@ -243,19 +243,24 @@ void expand_env_esc(char_u *srcp, char_u *dst, int dstlen, bool esc, bool one,
         // Verify that we have found the end of a UNIX ${VAR} style variable
         if (src[1] == '{' && *tail != '}') {
           var = NULL;
-        } else if (src[1] == '{') {
-          ++tail;
-        }
+        } else {
+          if (src[1] == '{') {
```

Comment:
It's not entirely the same: https://github.com/neovim/neovim/pull/2250/files#diff-145b92bbfb361cae1385b1efb2dea428R259